src/annotate.ts

```diff
@@ -43,43 +54,48 @@ export default async function annotate({payload, _ajv = ajv, _schema = schema}:
     json: true
   };
 
-  let json: any = await _request(options)
+  const json: any = await _request(options)
     .then((res: any) => {
       return res;
     })
     .catch((err: any) => {
-      //console.error('_request error: ' + err);
       throw new Error('_request error: ' + err);
     });
 
-  json = json[0];
-  //console.log(JSON.stringify(json, undefined, 2));
-  // if there is no match, returns undefined
-  // if there are multiple matches, returns only the first one.. check this is appropriate.
+  // if there is no match, returns [undefined]
+  // if there are multiple matches, returns all.
 
-  if (json) {
+  if (json.length > 0 && json[0] !== undefined) {
+    return json.map((x: any) => {
+      const uriSplit = x._links.olslinks[0].semanticTag.split('/');
 
-    const uriSplit = json._links.olslinks[0].semanticTag.split('/');
+      // make ontologyShortName equal to ontology accession prefix
+      let short_name = uriSplit.slice(-1)[0].split('_')[0].toLowerCase();
+      if (['format', 'operation', 'topic'].indexOf(short_name) > -1) short_name = 'edam';
+      if (short_name === 'orphanet') short_name = 'ordo';
 
-    // make ontologyShortName equal to ontology accession prefix
-    let ontologyShortName = uriSplit.slice(-1)[0].split('_')[0].toLowerCase();
-    if (['format', 'operation', 'topic'].indexOf(ontologyShortName) > -1) ontologyShortName = 'edam';
-    if (ontologyShortName === 'orphanet') ontologyShortName = 'ordo';
+      // const confidenceScoreF = R.cond([
+      //   [R.equals('HIGH'), R.always(1)],
+      //   [R.equals('GOOD'), R.always(0.75)],
+      //   [R.equals('MEDIUM'), R.always(0.5)],
+      //   [R.equals('LOW'), R.always(0.25)]
+      // ]);
 
-    const result = {
-      originalTerm: term,
-      ontologyTerm: json.annotatedProperty.propertyValue.toLowerCase(),
-      ontologyIRI: json._links.olslinks[0].semanticTag,
-      ontologyConfidence: json.confidence,
+      return {
+        term: x.annotatedProperty.propertyValue.toLowerCase(),
+        iri: x._links.olslinks[0].semanticTag,
+        confidence: confidenceScoreF(x.confidence),
```

Comment:
Is this property going to be an object or a single scalar value?

Reply:
@mareq asked for the object structure I've returned from the confidenceScoreF function